compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -2529,12 +2533,28 @@ LogicalResult initGPULaunchConfig(FunctionOpInterface funcOp) {
         break;
       }
     }
+
+    if (auto scatterOp = dyn_cast<IREE::LinalgExt::ScatterOp>(op)) {
+      Value indices = scatterOp.getIndices();
+      if (!indices.getDefiningOp()) {
+        continue;
+      }
+      // If scatter's indices is a generic op, mark it as to skip.
+      if (auto genericOp =
+              dyn_cast<linalg::GenericOp>(indices.getDefiningOp())) {
+        genericToSkip.insert(genericOp);
+      }
```

Comment:
You can add
```
  BackwardSliceOptions options;
  options.inclusive = true;
```
to `getBackwardSlice` so that you don't need to add the generics here and it will include them in the program slice.

Reply:
Good point, thanks for the suggestion!